frontend/public/module/auth.js

```diff
@@ -64,6 +61,54 @@ export const authSvc = {
       });
   },
 
+  // Extra steps are needed if this is OpenShift to delete the user's access
+  // token and logout the kube:admin user.
+  logoutOpenShift: (isKubeAdmin = false) => {
+    return authSvc.deleteOpenShiftToken().then(() => {
+      if (isKubeAdmin) {
+        authSvc.logoutKubeAdmin();
+      } else {
+        authSvc.logout();
+      }
+    });
+  },
+
+  deleteOpenShiftToken: () => {
+    return coFetch('/api/openshift/delete-token', { method: 'POST' })
+      // eslint-disable-next-line no-console
+      .catch(e => console.error('Error deleting token', e));
+  },
+
+  // The kube:admin user has a special logout flow. The OAuth server has a
+  // session cookie that must be cleared by POSTing to the kube:admin logout
+  // endpoint, otherwise the user will be logged in again immediately after
+  // logging out.
+  logoutKubeAdmin: () => {
+    clearLocalStorage();
+    // First POST to the console server to clear the console session cookie.
+    coFetch(window.SERVER_FLAGS.logoutURL, { method: 'POST' })
+      // eslint-disable-next-line no-console
+      .catch(e => console.error('Error logging out', e))
```

Comment:
We are good just logging this error, no handling?

Reply:
If the access token is deleted in the previous request, it's actually OK if this one fails since the console cookie will no longer have a valid token anyway. We might even be able to skip this step.

FWIW, this matches what we do in `logout()`

https://github.com/openshift/console/blob/master/frontend/public/module/auth.js#L55-L57

There is an edge case where both delete token and logout fail where it could be a problem, though. I guess that could happen if you were disconnected from the internet or the console pod goes away.